runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -312,9 +312,21 @@ public synchronized ILogData previous() {
         context.readQueue.clear();
 
         // If still null, we're done.
-        if (prevAddress == null) {
+        if (prevAddress == null && context.checkpointSuccessId == null) {
+            // The stream hasn't been checkpointed and we need to
+            // move the stream pointer to an address before the first
+            // entry
+            log.trace("previous[{}]: reached the beginning of the stream resetting" +
+                    " the stream pointer to {}", this, Address.NON_ADDRESS);
+            context.globalPointer = Address.NON_ADDRESS;
             return null;
+        } else if (prevAddress == null && context.resolvedQueue.first() == context.globalPointer) {
```

Comment:
Please don't use else here, it's not needed, because you return the result in prev condition

Reply:
done.